src/components/Confirm/index.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Drop } from 'grommet';
+import { Button, Box } from '../../';
+
+const defaultProps = {
+	align: {
+		bottom: 'top',
+	},
+};
+
+const Confirm = (props: any) => {
+	const { text, targetRef, onOK, onCancel, ...rest }: any = {
+		...defaultProps,
+		...props,
+	};
+
+	return (
+		open &&
+		targetRef.current && (
+			<Drop
+				{...rest}
+				target={targetRef.current}
+				onClickOutside={onCancel}
+				onEsc={onCancel}
+			>
+				<Box mx={2} mt={2} fontSize={3}>
+					{text}
+				</Box>
+				<Box>
+					<Button m={2} size="small" onClick={onCancel}>
+						Cancel
+					</Button>
+					<Button m={2} size="small" primary onClick={onOK}>
+						OK
+					</Button>
+				</Box>
+			</Drop>
+		)
+	);
+};
+
+export const useForwardedRef = (ref: any) => {
+	const innerRef = React.useRef(null);
+
+	React.useEffect(() => {
+		if (!ref) {
+			return;
+		}
+
+		if (typeof ref === 'function') {
+			ref(innerRef.current);
+		} else {
+			ref.current = innerRef.current;
+		}
+	});
+
+	return innerRef;
+};
+
+export const withConfirm = (Base: React.ComponentType) => {
+	return React.forwardRef(
+		({ onClick, confirmProps, ...rest }: any, ref: any) => {
+			const targetRef = useForwardedRef(ref);
```

Comment:
Why do we need `useForwardedRef` here, why can't we just use `ref`?  I guess because it can be undefined? Maybe `React.useRef(ref)` would work?

Reply:
> Why do we need `useForwardedRef` here, why can't we just use `ref`?  I guess because it can be undefined?

Yes.

> Maybe React.useRef(ref) would work?

I don't think just `useRef` will work. Ref is a regular object `{ current: ... }`, so `React.useRef({ current: ... })` will just result in `ref.current.current`.